pkg/infra/usagestats/statscollector/prometheus_flavor.go

```diff
@@ -0,0 +1,129 @@
+package statscollector
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+const promFlavorCacheLifetime = time.Hour
+
+type memoPrometheusFlavor struct {
+	variants map[string]int64
+
+	memoized time.Time
+}
+
+func (s *Service) detectPrometheusVariants(ctx context.Context) (map[string]int64, error) {
+	if s.promFlavorCache.memoized.Add(promFlavorCacheLifetime).After(time.Now()) &&
+		s.promFlavorCache.variants != nil {
+		return s.promFlavorCache.variants, nil
+	}
+
+	dsProm := &models.GetDataSourcesByTypeQuery{Type: "prometheus"}
+	err := s.datasources.GetDataSourcesByType(ctx, dsProm)
+	if err != nil {
+		s.log.Error("Failed to read all Prometheus data sources", "error", err)
+		return nil, err
+	}
+	group, ctx := errgroup.WithContext(ctx)
```

Comment:
This means if One of the requests fails all goroutines will be cancelled. I dont think thats what we want. 
In case we have 10 datasources and #1 is missconfigured we would loose the stats from the other datasources. A failing requests here should just be an noop IMO.

Reply:
I don't return errors on expected misconfigurations for this reason, that's only logged.

Reply:
:+1: